azure-pipelines.yml

```
@@ -3,19 +3,71 @@
 
 # Build STL targeting x86, x64, arm, arm64
 
-jobs:
-- template: azure-devops/run-build.yml
-  parameters:
-    targetPlatform: x86
+variables:
+  agentPool: 'StlBuild-2020-03-28'
+  vcpkgLocation: '$(Build.SourcesDirectory)/vcpkg'
 
-- template: azure-devops/run-build.yml
-  parameters:
-    targetPlatform: x64
+stages:
+  - stage: Code_Format_Validation
+    displayName: "Code Format Validation"
+    jobs:
+      - job: Run_Code_Format_Validation
+        timeoutInMinutes: 90
+        displayName: "Run Validation"
+        pool:
+          name: $(agentPool)
 
-- template: azure-devops/run-build.yml
-  parameters:
-    targetPlatform: arm
+        steps:
+          - checkout: self
+            clean: true
+            submodules: true
+          # TRANSITION: Once issue #19 or #33 is resolved on https://github.com/lukka/CppBuildTasks we can drop
+          # building boost-build here.
+          - task: run-vcpkg@0
+            displayName: 'Setup environment variables for run-cmake'
+            timeoutInMinutes: 10
+            inputs:
+              setupOnly: true
+              vcpkgArguments: 'boost-build'
+              vcpkgDirectory: '$(vcpkgLocation)'
+              vcpkgTriplet: 'x64-windows'
+          - task: run-cmake@0
+            displayName: 'Build Support Tools'
+            timeoutInMinutes: 2
+            inputs:
+              cmakeListsTxtPath: 'tools/CMakeSettings.json'
+              useVcpkgToolchainFile: true
+              configurationRegexFilter: '.*x64-Release.*'
+              buildDirectory: $(Build.ArtifactStagingDirectory)/tools
+          - task: BatchScript@1
+            displayName: 'Enforce clang-format'
+            timeoutInMinutes: 60
+            inputs:
+              filename: 'azure-devops/enforce-clang-format.cmd'
+              failOnStandardError: true
+              arguments: '$(Build.ArtifactStagingDirectory)/tools/parallelize/parallelize.exe'
+          - task: BatchScript@1
+            displayName: 'Validate Files'
+            timeoutInMinutes: 2
+            inputs:
+              filename: 'azure-devops/validate-files.cmd'
+              failOnStandardError: true
+              arguments: '$(Build.ArtifactStagingDirectory)/tools/validate/validate.exe'
+  - stage: Build_And_Test
+    displayName: "Build and Test the STL"

```

"the STL" part is quite obvious so I would suggest just this:
```suggestion
    displayName: "Build and Test"
```